### PR TITLE
misc changes

### PR DESCRIPTION
**Pull request by fanatic75**
allow filtering by providerType, userId in subscribers. Adds cancelled status for message status. Validate targets when creating message. delete all targets when user is deleted. fix twilio bug. add db env vars in messaging worker compose.phtml

## What does this PR do?

Review changes required 

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/config/errors.php [hunk 1]
```diff
@@ -847,5 +847,20 @@
...
+    Exception::MESSAGE_TARGET_NOT_SMS => [
+        'name' => Exception::MESSAGE_TARGET_NOT_SMS,
+        'description' => 'Message with the target ID is not an SMS target:',
+        'code' => 400,
+    ],
+    Exception::MESSAGE_TARGET_NOT_PUSH => [
+        'name' => Exception::MESSAGE_TARGET_NOT_PUSH,
+        'description' => 'Message with the target ID is not a Push target:',
+        'code' => 400,
+    ],
```
Comment (abnegate) [inline app/config/errors.php:864]: Let's use lower case for the target types (except SMS): email target, SMS target etc.

File: app/controllers/api/messaging.php [hunk 2]
```diff
@@ -2223,7 +2231,7 @@
     ->param('users', [], new ArrayList(new Text(Database::LENGTH_KEY), 1), 'List of User IDs.', true)
     ->param('targets', [], new ArrayList(new Text(Database::LENGTH_KEY), 1), 'List of Targets IDs.', true)
     ->param('description', '', new Text(256), 'Description for message.', true)
-    ->param('status', 'processing', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or processing.', true)
+    ->param('status', 'processing', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or cancelled or processing.', true)
```
Comment (abnegate) [inline app/controllers/api/messaging.php:2234]: draft, cancelled, or processing

Comment (fanatic75) [reply app/controllers/api/messaging.php]: fixed this

File: app/config/errors.php (hunk 1 above)
Comment (fanatic75) [reply app/config/errors.php]: updated this

**Review by abnegate — APPROVED** [review]

**Pull request merged**